finetrainers/trainer.py

```diff
@@ -864,11 +894,11 @@ def validate(self, step: int) -> None:
         logger.info(f"Memory after validation end: {json.dumps(memory_statistics, indent=4)}")
         torch.cuda.reset_peak_memory_stats(accelerator.device)
 
-        self.transformer.train()
+        if not final_validation:
```

Comment:
The reason for not performing final validation so far was to do something about it in `evaluate`. But anyway, these changes LGTM and we can think about evaluate later